xmtp_mls/src/subscriptions/stream_messages.rs

```diff
@@ -434,31 +434,6 @@ where
         });
     }
 
-    // iterative skip to avoid overflowing the stack
```

Comment:
I do like removing the `skip` function. That was a lot of surface area for things to be subtly different in how we process messages.